exporter/awsemfexporter/grouped_metric.go

```diff
@@ -62,6 +63,17 @@ func addToGroupedMetric(pmd *pdata.Metric, groupedMetrics map[interface{}]*group
 			}
 		}
 
+		// if patterns were found in config file and weren't replaced by resource attributes, replace those patterns with metric labels.
+		// if patterns are provided for a valid key and that key doesn't exist in the resource attributes, it is replaced with `undefined`.
+		if !patternReplaceSucceeded {
+			if strings.Contains(metadata.logGroup, "undefined") {
+				metadata.logGroup, _ = replacePatterns(config.LogGroupName, labels, config.logger)
+			}
+			if strings.Contains(metadata.logStream, "undefined") {
+				metadata.logStream, _ = replacePatterns(config.LogStreamName, labels, config.logger)
+			}
+		}
+
```

Comment:
could you help to get the benchmark test result with this change? This change make the log group check and replacement on every single metric data point in the critical path. I wonder what's the performance impact with this change.

Reply:
Thats a good point. I will do that.

Reply:
Also, I am not sure if replace the log group / stream by metric dimension is a good idea or not. Should we only allow to replace it by resource attributes. Can you move pod name to resource attribute level to avoid such changes?

Reply:
Hi Min, here are my thoughts. If customer have the values as resource attributes, they can do it using our existing approach. However, for customers who don't have a way to get this value from resource attributes, they need this feature. For your two points-

1. I am not sure why would it be a behavior change. Its an feature extension. Earlier, customers didn't have a way to choose the placeholder from metric labels, now they can. We are not changing existing behavior.

1. I don't think it will generate unexpected results. If customers put the placeholder for Pod name which is a requirement for EKS Fargate Container Inisghts, they will see as many different values as the number of Pods. 

Now the final question, can we move the PodName from metric label to resource attributes? If we want to do that, we need to do unnecessary conversion twice. 1) convert metric label to resource attributes in some kind of processor, and 2) convert the resource attributes back to metric labels in the awsemf exporter as we generate our dimensions from metric labels. No matter where we are doing these, but it will require some unnecessary calculation. However, if just allow this feature, we don't need any kind of extra calculations as we are going through all the steps for other purposes, like supporting EKS Fargate object generation. It just adding an extra check. Maybe I can share the benchmark and see how it performs. 
Waiting for your second thought. Thanks.

Reply:
Hi Min, I added three different benchmarks and didn't notice any extra burnout. I think its because of we were already going through the similar path for other calculations. 

i. Benchmark without any kind of pattern replacement.
ii. Benchmark where we only with Resource Attributes.
iii. Benchmark where we only replace with Metric Labels.

Here are results for two different runs.

First run:
```
BenchmarkTranslateOtToGroupedMetricWithoutConfigReplacePattern-4     	   33004	     35968 ns/op
BenchmarkTranslateOtToGroupedMetricWithConfigReplaceWithResource-4   	   32922	     36302 ns/op
BenchmarkTranslateOtToGroupedMetricWithConfigReplaceWithLabel-4      	   32852	     35511 ns/op
```
Second run:
```
BenchmarkTranslateOtToGroupedMetricWithoutConfigReplacePattern-4     	   33432	     36197 ns/op
BenchmarkTranslateOtToGroupedMetricWithConfigReplaceWithResource-4   	   33176	     35787 ns/op
BenchmarkTranslateOtToGroupedMetricWithConfigReplaceWithLabel-4      	   33133	     36192 ns/op
```